src/pages/transactions/[id].js

```diff
@@ -1,14 +1,24 @@
 import { useRouter } from "next/router";
+import { useState } from "react";
 import transactions from '../../data/transactions.json';
 import Link from "next/link";
 import ChevronLeft from "@/src/assets/icons/chevron_left.svg";
+import DeleteBin from "src/assets/icons/delete.svg"
 
-export default function TransactionDetails() {
+export default function TransactionDetails({handleDelete}) {
   const router = useRouter();
   const { id } = router.query;
+  const [modal, setModal] = useState(false);
+  const [deleteId, setDeleteId] = useState(null);
```

Comment:
this state should not be necessary, the `deleteId` will always be the `transaction.id`